### PR TITLE
completed user-can-favorite-locations feature.

### LINKED ISSUE #2: Feature: User can favorite location(s)
## ⚙️ Feature

> ##### What capability will users have when this feature is merged?

User can favorite location(s)
## 📐 Functionality

> ##### A description of how the application will work from the perspective of the user.

User can favorite locations to conveniently view them. 
## 🖼️ Wireframes

> ##### Wireframe screenshot(s) of what the app should look like when it is complete.

Screen Shot 2024-06-18 at 11 50 59 PM

## ✅ Task List

> ##### Commit checklist for tasks needed to complete the feature.

- [ ] ... 
- [ ] ...
- [ ] Make sure favorite icon and favorite location sections are responsive and looks good on both mobile and desktop sized screens. 

- [ ] Event Handling
   - [ ] Listen for clicks on the parent element of all rendered locations
   - [ ] Toggle the favorite location if the favorite icon was clicked
   - [ ] update favorite icon to visually show that location was favorited (red heart) 
 - [ ] Data Model
    - [ ] Update data model to include a favorites arrays for storing favorite locations
    - [ ] create local storage to save favorite locations in data.ts
    - [ ] in `data.js`, add a `'beforeunload'` event listener to the `window` object that serializes the data model as JSON and stores the JSON in `localStorage`.
- [ ] At the bottom of `data.js`, include a conditional statement that:
  - [ ] checks to see if there is a JSON data model stored in `localStorage`.
  - [ ] if there is, parse it and assign it to the initial `data` variable so that the user can continue adding entries to the pre-existing data model.
          - [ ]   attached event listener to object and listen for the beforeunload event
          - [ ] create callback function to stringify the favorites array 
          - [ ] store key/value pair to localStorage containing the JSON string
          - [ ] access localStroage using key name and assign the return result to a variable

- [ ] Form submission for new locations
     - [ ] Make sure form's input values are used to fetch weather data for a new location
     - [ ] add the new location to the rendered locations list and the data model
     - [ ] include the favorite icon for the new location
-
- [ ] Make sure that click targets are either `<a>` or `<button>` elements.
- [ ] Make sure that all click targets show a `cursor: pointer` when hovered with the mouse.
- **Clean Up**
  - [ ] Remove all buttons and placeholders that do not serve the functionality of the feature.
  - [ ] Remove all console logs.
  - [ ] Remove all commented out code.
  - [ ] Remove all CSS properties that have no effect.
  - [ ] Check all code for proper formatting and indentation.
  - [ ] Confirm that there are no errors in the console while using the application.
  - [ ] Confirm that all previous functionality still works without errors.
  - [ ] Confirm that the user interface looks natural on both mobile and desktop screens.
  - [ ] Create GIFs of the feature being used on both mobile and desktop screen sizes for inclusion in Pull Request.

### PR DESCRIPTION
**Pull request by Katelyn-Ngo**
Recording 2024-06-24 at 19 17 22

sorry i just realized theres also commented out css that i should of delete will delete when i submit the PR again!!

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -30,25 +30,32 @@
...
         </div>
 
         <!-- Weather Display View -->
         <div class="hidden" data-view="weather-display">
           <div class="row top-section">
             <a href="#" id="back-button"> Back </a>
+            <!-- <a href="#" id="favorites-button3" -->
+            <!-- ><i class="fa-solid fa-heart"></i
+            ></a> -->
+
```
Comment (brandon-moy) [inline index.html:49]: This can be removed

File: css/styles.css [hunk 2]
```diff
@@ -38,7 +52,7 @@ button:hover {
   height: 45px;
   margin-right: 10px;
   padding-top: 10px;
-  padding-left: 6px;
+  padding-left: 6px; */
```
Comment (brandon-moy) [inline css/styles.css:55]: This is likely the CSS you mentioned in your comment.

**Comment by Katelyn-Ngo** [comment]
Recording 2024-06-25 at 09 20 52

**Review by brandon-moy — APPROVED** [review]
Approved! ✅

**Pull request merged**